### PR TITLE
cgal 4.8.1

### PR DESCRIPTION
**Pull request by dunn**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
  Closes #195.
- [x] Does your submission pass `brew audit --strict --online <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Have you built your formula locally prior to submission with `brew install <formula>`?

cc @Homebrew/science for revision bumps.

### CODE REVIEW

**Comment by dunn** [comment]
Huh, I saw a similar error locally with pgrouting but it didn't cause the test to fail.

**Comment by dunn** [comment]
Also weird that it's only on El Capitan.

**Comment by dunn** [comment]
When I run the test commands outside of Homebrew I get the `malloc` error before getting to `pgrouting`; it happens at `CREATE EXTENSION postgis;`.

**Comment by dunn** [comment]
And accordingly, the `postgis` test is failing for me locally:
```
🌲  brew test -v postgis
Testing postgis
==> Using the sandbox
/usr/bin/sandbox-exec -f /tmp/homebrew20160414-47408-4nzzda.sb /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/bin/ruby -W0 -I /usr/local/Library/Homebrew -- /usr/local/Library/Homebrew/test.rb /usr/local/Library/Taps/homebrew/homebrew-core/Formula/postgis.rb -v
==> /usr/local/Cellar/postgis/2.2.1/bin/shp2pgsql /private/tmp/postgis20160414-47409-59ba5g/brew.shp
Shapefile type: PointZ
Postgis type: POINT[4]
shp2pgsql(47417,0x7fff7f69b000) malloc: *** error for object 0x1: pointer being freed was not allocated
*** set a breakpoint in malloc_error_break to debug
Error: postgis: failed
<0> expected but was
<nil>.

**Comment by sjackman** [comment]
@Homebrew/science `biocgal`, `dgtal` and `graph-tool` depend on `cgal`.

**Comment by apjanke** [comment]
Based on the behavior here, and what I saw when I was testing #195, this sounds like an actual wild pointer bug in this release of `cgal`, or some linked package which was updated since the last `cgal` release. Any differences in system configuration could be shifting memory layout around and affecting when the bug gets triggered. And since the killed process is inside the server, that malloc/free crash might not cause the test to fail, depending on how much work it got done before crashing.

**Comment by dunn** [comment]
Using the new postgis didn't help so you're probably right.

**Comment by dunn** [comment]
Ack, sorry for letting this sit.

I don't think I understand the error well enough to file a good bug report.  Would @apjanke or someone else be willing to poke upstream about this? 😽

**Comment by apjanke** [comment]
I've got a lot on my plate at the moment. I wouldn't be able to do much debugging of this anyway; pointer problems are hard to reproduce. I wouldn't even know which upstream - cgal or postgis or postgres - to report it to.

All a good bug report would need is "I installed it using these exact steps, ran this little script to test it, and it crashes with this error message". And then you can try sending it to all of them and see who bites.

If I have more free time next week I'll see if I can take a stab at it.

**Comment by drewish** [comment]
One thing that came up in https://github.com/Homebrew/legacy-homebrew/commit/793caf9c496a8c113b6341c96faba8b7c5be8246 is that we should be requiring qt5. Not sure if we want to include that change in this PR.

**Comment by mikemcquaid** [comment]
If we can use `qt5` instead of `qt` then 👍 at this point.

**Comment by DomT4** [comment]
Apparently the only failure remaining is `pgrouting`'s test on El Cap.

**Comment by drewish** [comment]
They've got a 4.8.1 release out now, do we want to update this to that and see if it magically fixes any of the test failures?

**Comment by lrineau** [comment]
Hi, I am the release manager of the CGAL project. I know that the change set from 4.8 to 4.8.1 is rather small, but try to update anyway.

I can see that you get a malloc error. Is there a way to reproduce the issue, and maybe try to reduce the test case?

File: Formula/cgal.rb [hunk 1]
```diff
@@ -11,29 +11,25 @@ class Cgal < Formula
...
   deprecated_option "imaging" => "with-imaging"
 
+  option :cxx11
   option "with-imaging", "Build ImageIO and QT compoments of CGAL"
   option "with-eigen3", "Build with Eigen3 support"
   option "with-lapack", "Build with LAPACK support"
 
   depends_on "cmake" => :build
+  depends_on "mpfr"
+  depends_on "qt5" if build.with? "imaging"
```
Comment (MikeMcQuaid) [inline Formula/cgal.rb]: Would be good to migrate this (and the option below) to `:optional`s

**Comment by dunn** [comment]
@lrineau here's the failure: http://bot.brew.sh/job/Homebrew%20Core%20Pull%20Requests/4922/version=el_capitan/testReport/junit/brew-test-bot/el_capitan/test_pgrouting/

You can pull down this branch and try `brew test -v --sandbox cgal` and see if it reproduces for you locally.

**Comment by MikeMcQuaid** [comment]
@dunn Any news here?

**Comment by dunn** [comment]
No :(

Should I close, since this is an upstream problem?

**Comment by MikeMcQuaid** [comment]
Yeh :(

**Comment by ilovezfs** [comment]
The problem here turned out to be that sfcgal needed to be rebuilt against the new cgal.

See #4290
@lrineau I've also tested CGAL-4.9-beta1 and it works too. Any idea when 4.9 will ship?

**Comment by lrineau** [comment]
4.9 will be around mid-september. I am the release manager. I should know.

**Comment by ilovezfs** [comment]
@lrineau cool thanks!

**Pull request closed without merging**